### PR TITLE
Feature/broadcast pacman

### LINKED ISSUE #83: Create playerId
Create playerId when user joins the room.
Add ids to players when they join the room. 
Start game after 3 seconds when two players join the room.
Emit new gameState after every 200ms.
Move ghost logic to server.

### PR DESCRIPTION
**Pull request by Shalinit3**
Will setup babelrc.
Will add ids to players when they join the room. 
Will start game after 3 seconds when two players join the room.
Will emit new gameState after every 200ms(current refreshRate).
Will move ghost logic to server.

Note: consoles are for testing till we use those values in client. 

resolves: #83

### CODE REVIEW

File: server/api/game/controller.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  const ghostsUpdated = ghosts.map(
+    ({ x, y, direction }) => getRandomAdjacentAvailableCell(gridState, { x, y, direction }),
+  );
+  return {
+    ghostsUpdated, moveGhostsCount,
+  };
+};
+
+const animateGame = () => {
```
Comment (shivangg) [inline server/api/game/controller.js]: animate in something that the client will do. Change this to something like `calculateNextGameState()`

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 2]
```diff
@@ -62,7 +63,8 @@ class PacmanGame extends Component {
 
   startGame = () => {
     const { config, pacman } = this.state;
-    joinGame(pacman);
+    joinGame({ playerId: uuid.v1(), ...pacman });
```
Comment (nimish-gupta) [inline client/src/components/PacmanGame/PacmanGame.jsx:66]: PlayerId can be userId of the authenticated users

File: server/api/game/constants.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const directionValues = {
+  UP: { x: 0, y: -1 },
+  DOWN: { x: 0, y: 1 },
+  RIGHT: { x: 1, y: 0 },
+  LEFT: { x: -1, y: 0 },
+};
+
+const getGhosts = () => [[13, 12, 'RIGHT'], [14, 12, 'LEFT'], [11, 12, 'DOWN'], [12, 12, 'UP']];
+
+const boardCorners = [{ x: 1, y: 1 }, { x: 1, y: 23 }, { x: 23, y: 1 }, { x: 23, y: 23 }];
```
[the remark this answers is not in the document]
Comment (nimish-gupta) [reply server/api/game/constants.js]: Board corners can be detected using board, finstead of hardcoding.

File: server/api/game/constants.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+const refreshRate = 200;
+
+const directionValues = {
+  UP: { x: 0, y: -1 },
+  DOWN: { x: 0, y: 1 },
+  RIGHT: { x: 1, y: 0 },
+  LEFT: { x: -1, y: 0 },
+};
+
+const getGhosts = () => [[13, 12, 'RIGHT'], [14, 12, 'LEFT'], [11, 12, 'DOWN'], [12, 12, 'UP']];
```
Comment (nimish-gupta) [inline server/api/game/constants.js]: This can be randomly generated.

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 5]
```diff
@@ -124,6 +126,7 @@ class PacmanGame extends Component {
 
   setGameStatus = (status) => {
     if (status === 'finish') {
+      leaveGame();
       clearInterval(this.animationHandler);
       this.setState({ status: 2 });
```
Comment (nimish-gupta) [inline client/src/components/PacmanGame/PacmanGame.jsx:131]: Set status values as constants, by this we can know which number correspond to which status

File: server/api/game/constants.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    2: 'energizer',
+    3: 'ghost',
+    4: 'wall',
+    5: 'pacman',
+  };
+  return entityMap[code];
+};
+
+const entityToCode = (entity) => {
+  const entityMap = {
```
Comment (nimish-gupta) [inline server/api/game/constants.js:55]: You can create this with the help of above entity map, you just have to swap key/values

File: server/api/game/controller.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
+const uuidv1 = require('uuid/v1');
```
Comment (nimish-gupta) [inline server/api/game/controller.js]: Have you guys not setup the babel? If no please set it up, you can then use import/export es6 features

File: server/api/game/controller.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    arr.push(boardCorners[index]);
+    return addPositionsToArray(arr, index);
+  }
+  return arr;
+};
+
+const moveGhosts = ({ ghosts, gridState, scatterStart }) => {
+  let { moveGhostsCount, scatterGhostspath } = gameState;
+  moveGhostsCount += 1;
+  const scatterEnd = scatterStart + 55;
```
Comment (nimish-gupta) [inline server/api/game/controller.js:57]: What is 55?

File: server/index.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ require('dotenv/config');
 
 const app = express();
 const server = http.Server(app);
-const io = require('socket.io')(server);
+global.io = require('socket.io')(server);
```
Comment (nimish-gupta) [inline server/index.js]: Why global?

**Review by nimish-gupta — CHANGES_REQUESTED** [review]
Please ads babel. Also add tests for the server side logic functions. Apart from that gansta PR :)

File: server/index.js (hunk 9 above)
Comment (Shalinit3) [reply server/index.js]: I am using it in socketService file so made it gobal.

File: client/src/components/PacmanGame/PacmanGame.jsx (hunk 2 above)
Comment (Shalinit3) [reply client/src/components/PacmanGame/PacmanGame.jsx:66]: Yes, we will be doing that. Its temporary.

File: server/api/game/constants.js (hunk 4 above)
Comment (kaushikb11) [reply server/api/game/constants.js]: Ghosts are generated from the "monster pen" which is around the center of the board. So, had to set it like that.

File: client/src/components/PacmanGame/PacmanGame.jsx (hunk 5 above)
Comment (kaushikb11) [reply client/src/components/PacmanGame/PacmanGame.jsx:131]: Planning to add more statues in my next PR, to pause, resume and restart the game.

File: server/api/game/controller.js (hunk 8 above)
Comment (kaushikb11) [reply server/api/game/controller.js:57]: That's the duration of the scatter mode. The time for which the Ghosts spread out on the board, so for 55 * refreshrate = t sec.

Comment (kaushikb11) [reply server/api/game/controller.js:57]: Will assign a variable for better understanding

File: server/api/game/constants.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaushikb11) [reply server/api/game/constants.js]: My bad. Will make it dependent on the board size.

File: server/api/game/controller.js (hunk 7 above)
Comment (Shalinit3) [reply server/api/game/controller.js]: Added babelrc and updated the server code.

**Review by nimish-gupta — APPROVED** [review]

**Pull request merged**